packages/quantic/force-app/main/default/lwc/quanticNumericFacet/e2e/quanticNumericFacet.e2e.ts

```diff
@@ -0,0 +1,139 @@
+import {testSearch, testInsight, expect} from './fixture';
+import {
+  useCaseEnum,
+  useCaseTestCases,
+} from '../../../../../../playwright/utils/useCase';
+import facetData from './data';
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+  test.describe(`quantic numeric facet ${useCase.label}`, () => {
+    test.describe('when selecting and deselecting a facet value', () => {
+      test('should trigger a new search and log the corresponding UA analytics events', async ({
+        baseFacet,
+        facet,
+      }) => {
+        const selectedIndex = 0;
+        const {facetId, field, values} = facetData;
+        const facetSelectUaRequest = baseFacet.waitForFacetSelectUaAnalytics({
+          facetId,
+          facetField: field,
+          facetValue: `${values[selectedIndex].start}..${values[selectedIndex].end}`,
+        });
+        let searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetSelectUaRequest;
+        let searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+
+        const facetDeselectUaRequest =
+          baseFacet.waitForFacetDeselectUaAnalytics({
+            facetId,
+            facetField: field,
+            facetValue: `${values[selectedIndex].start}..${values[selectedIndex].end}`,
+          });
+        searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetDeselectUaRequest;
+        searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetDeselect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetDeselect.actionCause).toEqual('facetDeselect');
+      });
+    });
+
+    test.describe('when selecting a facet value and clicking the clear filter button', () => {
+      test('should trigger a new search and log the corresponding UA analytics events', async ({
+        baseFacet,
+        facet,
+      }) => {
+        const selectedIndex = 0;
+        const {facetId, field, values} = facetData;
+        const facetSelectUaRequest = baseFacet.waitForFacetSelectUaAnalytics({
+          facetId,
+          facetField: field,
+          facetValue: `${values[selectedIndex].start}..${values[selectedIndex].end}`,
+        });
+        let searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetSelectUaRequest;
+        let searchResponse = await searchResponsePromise;
```

Comment:
Same here.

Reply:
I can update if you don't prefer this pattern.